### DISCUSSION TITLE
Creating API Endpoints in Serverpod

### DISCUSSION THREAD

**Discussion by Sameer330** [serverpod/serverpod#1831] [Q&A] [answered]
Flutter Dev Here.

New to Serverpod! 🙌

I created my first project using a tutorial on YouTube. I wanted to test an endpoint in my project. 

I used Postman to test the `ExampleEndpoint` class from the boilerplate code. 

This is the Postman console log:

```
POST https://localhost:8080/example
Error: Client network socket disconnected before secure TLS connection was established

Request Headers
  Content-Type: application/json
  User-Agent: PostmanRuntime/7.36.1
  Accept: */*
  Cache-Control: no-cache
  Postman-Token: 1e66cdf6-2a16-4970-af87-fddb2542516a
  Host: localhost:8080
  Accept-Encoding: gzip, deflate, br
  Connection: keep-alive

Request Body
  {
      "method": "hello",
      "name": "Sameer"
  }
```

I tried calling the endpoint using the URL: `https://localhost:8080/example` but did not get the expected result.

I want the API to work similar to how the APIs work when running `npm run dev` in a JavaScript backend server.

How to make this work?

**Comment by abdelaziz-mahdy** [comment]
i had this problem too, but i was able to find the endpoints from the client code 
for example `benchmark_client/lib/src/protocol/client.dart`
<img width="620" alt="image" src="https://github.com/serverpod/serverpod/assets/25157308/1a6caa8b-9315-45ba-ab92-87bfa072e245">
the endpoints are 
<img width="477" alt="image" src="https://github.com/serverpod/serverpod/assets/25157308/8133a320-8669-4233-a56f-9d4f85ced8a1">



and i think this pr will help greatly in these use cases https://github.com/serverpod/serverpod/pull/1454

**Reply by Sameer330** [reply, marked as answer]
I am currently running Serverpod version 1.2.1.

Although I didn't find the exact code in my client, I put hit the following API: `http:localhost:8080/example/hello`

Earlier I was using `https` instead of `http` in my API call, which returned an error.

Thanks! 🙌
